File: campushub-frontend/src/services/attendanceService.js
import api from "../api/axiosConfig";

export const getAllAttendance = async () => {
    const response = await api.get("/attendance");
    return response.data.data;
};

export const createAttendance = async (attendance) => {
    const response = await api.post("/attendance", attendance);
    return response.data.data;
};

export const updateAttendance = async (id, attendance) => {
    const response = await api.put(`/attendance/${id}`, attendance);
    return response.data.data;
};

export const deleteAttendance = async (id) => {
    const response = await api.delete(`/attendance/${id}`);
    return response.data.data;
};

export const saveAttendance = async (attendanceList) => {
    await api.post("/attendance/bulk", attendanceList);
};

export const getAttendanceByCourseAndDate = async (
    courseId,
    attendanceDate
) => {

    const response = await api.get(
        `/attendance/course/${courseId}`,
        {
            params: {
                attendanceDate
            }
        }
    );

    return response.data.data;
};

File: src/main/java/com/preeti/campushub/service/AttendanceService.java
package com.preeti.campushub.service;

import java.time.LocalDate;
import java.util.List;

import com.preeti.campushub.dto.attendance.AttendanceRequest;
import com.preeti.campushub.dto.attendance.AttendanceResponse;

public interface AttendanceService {

    AttendanceResponse createAttendance(AttendanceRequest request);

    List<AttendanceResponse> getAllAttendance();

    AttendanceResponse getAttendanceById(Long id);

    AttendanceResponse updateAttendance(Long id, AttendanceRequest request);

    void deleteAttendance(Long id);

    List<AttendanceResponse> getMyAttendance();

    void saveAttendance(List<AttendanceRequest> requests);

    List<AttendanceResponse> getAttendanceByCourseAndDate(
        Long courseId,
        LocalDate attendanceDate);

}